File: samples/ios/KsDialogsSample/InlineDialogViewModel.swift
import KsDialogs

/// Inline Dialog の ViewModel。
///
/// 結果型を宣言していないため、結果は既定の真偽値になる (core/ADR-0012)。
/// レジストリには登録せず、表示のたびに中身を直接渡すため、この型が登録キーになることはない。
final class InlineDialogViewModel: DialogViewModel {
    /// ダイアログに表示するメッセージ。
    let message: String

    init(message: String) {
        self.message = message
    }
}

File: samples/android/app/src/main/kotlin/jp/kamusoft/ksdialogs/samples/android/InlineDialogViewModel.kt
package jp.kamusoft.ksdialogs.samples.android

import jp.kamusoft.ksdialogs.SimpleDialogViewModel

/**
 * Inline Dialog の ViewModel。
 *
 * 結果型を書かずに済む顔で宣言しているため、結果は真偽値になる (core/ADR-0012)。
 * レジストリには登録せず表示のたびに中身を直接渡すため、このクラスが登録キーになることはない。
 *
 * @property message ダイアログに表示するメッセージ
 */
internal class InlineDialogViewModel(val message: String) : SimpleDialogViewModel
